profiles/tests/test_gql_profiles_query.py

```diff
@@ -194,29 +194,55 @@ def test_staff_user_can_filter_profiles_by_a_field(
         assert executed["data"] == expected_data
 
 
-@pytest.mark.parametrize(
-    "field_name", ["first_name", "last_name", "national_identification_number"]
-)
-def test_staff_user_can_not_filter_profiles_by_verified_personal_information_without_required_permission(
-    field_name, user_gql_client, group, service
+@pytest.mark.parametrize("amr_claim_value", [None, 0, "helsinki_adfs", "foo"])
+@pytest.mark.parametrize("has_needed_permission", [True, False])
+def test_staff_user_filter_profiles_by_verified_personal_information_permissions(
+    has_needed_permission, amr_claim_value, settings, user_gql_client, group, service
```

Comment:
Above this test is a test called `test_staff_user_can_filter_profiles_by_a_field`. Then there was (before this change) a test called `test_staff_user_can_not_filter_profiles_by_verified_personal_information_without_required_permission`. The first one tested cases where filtering succeeds, the latter tested cases where filtering doesn't succeed.

Could that separation of concerns be preserved between these two tests? So the first test would check that filtering works when permissions, settings and amr-claims are all as needed and the second test would check that if any of those is not as needed, then filtering returns no results?

Reply:
This new function tests all of the cases when searching by the verified personal information. Are you saying we should not parametrize, but separate in two test functions?

Reply:
There were two test functions already before. Both of them can be parametrized as needed. I mean that the first test would check all the cases where the filtering returns some profiles (i.e. correct permissions are in place) and the second test would check all the cases where no profiles are returned (i.e. some permission or token claim is missing or incorrect).

Reply:
The existing `test_staff_user_can_filter_profiles_by_a_field` also filters by verified personal information, although it doesn't say so in its name.